iceoryx_dds/include/iceoryx_dds/gateway/iox_to_dds.hpp

```diff
@@ -34,14 +35,28 @@ template <typename channel_t = gw::Channel<popo::UntypedSubscriber, dds::data_wr
 class Iceoryx2DDSGateway : public gateway_t
 {
   public:
+    /// @brief Default constructor
     Iceoryx2DDSGateway() noexcept;
+
+    /// @brief Load the provided gateway configuration
+    /// @param[in] config The gateway configuration to load
     void loadConfiguration(const config::GatewayConfig& config) noexcept;
```

Comment:
What happens when it fails to load the configuration?

Reply:
There's no check for a valid config here. It's currently used in a successful `expected`.